### PR TITLE
[Issue #445] Update hasNewActivity indicator to show post number

### PR DESCRIPTION
**Pull request by peterphan1996**
## Status
**READY**

## Issue links
https://github.com/venturemark/webclient/issues/445

## Description
Update hasNewActivity indicator to show post number

## Screen recording
https://user-images.githubusercontent.com/28189578/145387412-4668ced8-77c3-46c3-a944-d469bc499a1f.mov

### CODE REVIEW

File: src/component/sidebaritem/index.tsx [hunk 1]
```diff
@@ -102,6 +104,19 @@ function SidebarItem(
...
+            <span
+              style={{
+                color: "#fff",
+                fontWeight: 500,
+              }}
+            >
+              {unreadCount}
+            </span>
+          ),
+        }}
```
Comment (peterphan1996) [inline src/component/sidebaritem/index.tsx:119]: Will improve by changing counterSlot prop to type `number`

**Comment by peterphan1996** [comment]
@tfussell As you can see in the screen recording, the unread count badge flashes. Do you think there's a immediate API call after rendering to update the `lastViewUpdate`?

**Comment by tfussell** [comment]
> @tfussell As you can see in the screen recording, the unread count badge flashes. Do you think there's a immediate API call after rendering to update the `lastViewUpdate`?

@peterphan1996 Yes, there's this effect hook in `FeedUpdate` https://github.com/venturemark/webclient/blob/82c8e1924aaf35bab86326c3c22dbae77abcb3df/src/component/feedupdate/index.tsx#L95-L147 which updates the `lastUpdate` on the current user in the API. The idea is that rendering the component implies that the user has viewed all of the updates in that timeline.

**Comment by tfussell** [comment]
I can't currently test this myself because add members is broken and I usually use a second member to test the unread indicator. Code makes sense to me.

File: src/component/sidebaritem/index.tsx (hunk 1 above)
Comment (tfussell) [reply src/component/sidebaritem/index.tsx:119]: Why can't we use the default Plasmic style here? Does it not look right?

Comment (peterphan1996) [reply src/component/sidebaritem/index.tsx:119]: counterSlot prop actually expects an element, I tried with `counterSlot={unreadCount}` but the style doesn't match. Maybe we need to update on Plasmic?

**Comment by peterphan1996** [comment]
> webclient/src/component/feedupdate/index.tsx

@tfussell Hmm that's weird, according to the screen recording, on the right browser, the 2nd venture that got new update didn't render at all, so how the unread count got cleared right away?

File: src/component/sidebaritem/index.tsx (hunk 1 above)
Comment (tfussell) [reply src/component/sidebaritem/index.tsx:119]: Makes sense. We can go with this for now then.

**Comment by tfussell** [comment]
> @tfussell Hmm that's weird, according to the screen recording, on the right browser, the 2nd venture that got new update didn't render at all, so how the unread count got cleared right away?

I'm taking a look at this.

**Comment by peterphan1996** [comment]
https://github.com/venturemark/webclient/blob/82c8e1924aaf35bab86326c3c22dbae77abcb3df/src/component/feedupdate/index.tsx#L132 Hi @tfussell, I'm not sure this will help but I think that this function might update those lastView data which causes the indicator flashing

**Comment by tfussell** [comment]
> https://github.com/venturemark/webclient/blob/82c8e1924aaf35bab86326c3c22dbae77abcb3df/src/component/feedupdate/index.tsx#L132
> Hi @tfussell, I'm not sure this will help but I think that this function might update those lastView data which causes the indicator flashing

`reset()` just allows the mutation to be run again so I'm not sure it's the problem. Adding `console.log({userLastUpdates, newLastUpdates})` here https://github.com/venturemark/webclient/blob/82c8e1924aaf35bab86326c3c22dbae77abcb3df/src/component/feedupdate/index.tsx#L119 should show if the correct values are being updated.

**Review by tfussell — APPROVED** [review]

**Review by tfussell — APPROVED** [review]
Confirmed this works on my machine.

**Pull request merged**